Applications/SlicerApp/Testing/Python/CMakeLists.txt

```diff
@@ -178,11 +178,15 @@ endif()
 # Check if scripted module import works as expected
 #
 
+set(testname ScriptedModuleDiscoveryTest)
 slicer_add_python_test(
-  SCRIPT ScriptedModuleDiscoveryTest.py
+  SCRIPT ${testname}.py
   SLICER_ARGS --no-main-window --disable-builtin-modules --additional-module-path ${CMAKE_CURRENT_SOURCE_DIR}/ScriptedModuleDiscoveryTest
   TESTNAME_PREFIX nomainwindow_
   )
+set_tests_properties(py_nomainwindow_${testname} PROPERTIES
+  PASS_REGULAR_EXPRESSION "ModuleAWidget finalized"
```

Comment:
This test will fail on Windows with console disabled. Instead of using stdout to communicate with the tester process, use output files instead, the same way as it is done for slicerrc and startupCompleted event testing.

Reply:
Good catch. I will fix it before the nightly kick in

>  create some common utilities to make it easier to implement module tests that write output files.

Indeed. I will look into that

Reply:
Waiting I implement a proper fix. I am excluding the text on windows